docs/examples/image_binarization/image_binarization.jl

```diff
@@ -0,0 +1,128 @@
+# ---
+# title: Image Binarization
+# author: Ashwani Rathee
+# date: 2021-7-2
+# ---
+
+# ImageBinarization.jl provides several algorithms for binarizing images
+# into background and foreground(bi-level image). In this demonstration, 
+# we'll be exploring where these algorithms could be useful and learn how
+# different algorithms are suited for different types of tasks.
+
+# Suppose a person wants to make an automatic Sudoku solver using Computer 
+# Vision . Before anything else that person needs to import the image, 
+# preprocess it to optimize it for cell content extraction which can then 
+# be used for solving the sudoku. 
+
+# Let's first import packages and then testimage sudoku
+
+using Images, ImageBinarization, TestImages
+using  CoordinateTransformations, Rotations, ImageTransformations
+
+# Original Image
+
+img = testimage("sudoku")
+
+# Grayscale version as binarize! only accepts Grayscale images
+
+img = Gray.(img)
+
+# Let's first rotate our image
+
+trfm = recenter(RotMatrix(pi / 30), center(img));
+imgw = warp(img, trfm)
+
+# Now let's zoom in a bit
+
+imgw = parent(imgw)[65:490, 65:490]
+
+# Now let's binarize an image using the `Sauvola` algorithm
+
+alg = Sauvola();
+img_otsu = binarize(imgw, alg)
+mosaicview(imgw, img_otsu; ncol = 2)
+
+# Now the differences between the binarized and non-binarized images 
+# become apparent. Let's now implement a function to see how different algorithms 
+# perform for this particular problem.
+
+algs = [
+    "AdaptiveThreshold",
+    "Balanced",
+    "Entropy",
+    "Intermodes",
+    "MinimumError",
+    "Moments",
+    "Niblack",
+    "Otsu",
+    "Polysegment",
+    "Sauvola",
+    "UnimodalRosin",
+    "Yen",
+]
+function binarize_methods(img_input, algs)
+    imgs_binarized = Array[]
+    for i in algs
+        alg = getfield(ImageBinarization, Symbol(i))()
+        img_input1 = binarize(img_input, alg)
+        push!(imgs_binarized, img_input1)
+    end
+    return imgs_binarized
+end
+
+output = binarize_methods(imgw, algs);
+
+
+for i = 1:12 #src
+    save("assets/test$(i).png", RGB.(output[i])) #src
+end #src
+
+# | Algorithm | Algorithm | Algorithm |
+# | :------: | :------: | :------: |
+# | AdaptiveThreshold | Balanced | Entropy |
+# | ![](assets/test1.png)  | ![](assets/test2.png)  | ![](assets/test3.png) |
+# | MinimumError |  Moments | Niblack |
+# | ![](assets/test5.png) | ![](assets/test6.png) | ![](assets/test7.png) |
+# | Polysegment | Sauvola | UnimodalRosin |
+# | ![](assets/test9.png) | ![](assets/test10.png) | ![](assets/test11.png) |
+# | Intermodes | Otsu | Yen |
+# | ![](assets/test4.png) | ![](assets/test8.png) | ![](assets/test12.png) |
+
+# We can choose one of methods based on the results here, and use `OCReact.jl` 
+# which is based on Tesseract OCR to find the content in the sudoku and then
+# solve the sudoku.
```

Comment:
I can't find `OCReact.jl` in JuliaHub and GitHub.

Reply:
https://github.com/leferrad/OCReract.jl extra r issue

Reply:
I think we should use https://github.com/pixel27/Tesseract.jl instead because it's backed by Tesseract_jll

Reply:
OCReract.jl is better maintained I think, no updates in Tesseract.jl in last 15 months. But https://github.com/JuliaPackaging/Yggdrasil shows that Tesseract folder was updated in past 12 days. I don't know enough binary builds, so we can go for Tesseract.jl if you think that's a better option

Reply:
> OCReract.jl is better maintained I think

Nope, `OCReract.jl` requires the user to manually install Tesseract and make it available in PATH. While Tesseract.jl ships a prebuilt Tesseract using BinaryBuilder so that user just need `pkg> add Tesseract` and that's all.